tests/test_messages_queue.cc

```diff
@@ -0,0 +1,482 @@
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#include "../src/file_manager.h"
+#include "../src/messages_queue.h"
+
+#include <algorithm>
+#include <chrono>
+#include <cstdlib>
+#include <future>
+#include <iostream>
+#include <map>
+#include <random>
+#include <thread>
+#include <vector>
+
+#define TEST_EQUAL(x, y)                                                                                        \
+  {                                                                                                             \
+    auto vx = (x);                                                                                              \
+    auto vy = (y);                                                                                              \
+    if (vx != vy) {                                                                                             \
+      std::cerr << __FILE__ << ':' << __FUNCTION__ << ':' << __LINE__ << " Test failed: " << #x << " != " << #y \
+                << " (" << vx << " != " << vy << ")" << std::endl;                                              \
+      std::exit(-1);                                                                                            \
+    }                                                                                                           \
+  }
+
+struct ScopedRemoveFile {
+  std::string file_;
+  ScopedRemoveFile(const std::string & file) : file_(file) {}
+  ~ScopedRemoveFile() { std::remove(file_.c_str()); }
+};
+
+using alohalytics::FileManager;
+
+// Generates unique temporary file name or empty string on error.
+static std::string GenerateTemporaryFileName() {
+#ifdef _MSC_VER
+  char tmp_file[L_tmpnam];
+  if (0 == ::tmpnam_s(tmp_file, L_tmpnam)) {
+    return tmp_file;
+  }
+#else
+  char tmp_file[] = "/tmp/alohalytics_file_manager-XXXXXX";
+  if (::mktemp(tmp_file)) {
+    return tmp_file;
+  }
+#endif
+  return std::string();
+}
+
+void Test_GetDirectoryFromFilePath() {
+  const std::string s = std::string(1, FileManager::kDirectorySeparator);
+  const std::string ns = (s == "/") ? "\\" : "/";
+  TEST_EQUAL("", FileManager::GetDirectoryFromFilePath(""));
+  TEST_EQUAL(".", FileManager::GetDirectoryFromFilePath("some_file_name.ext"));
+  TEST_EQUAL(".", FileManager::GetDirectoryFromFilePath("evil" + ns + "file"));
+  TEST_EQUAL("dir" + s, FileManager::GetDirectoryFromFilePath("dir" + s + "file"));
+  TEST_EQUAL(s + "root" + s + "dir" + s, FileManager::GetDirectoryFromFilePath(s + "root" + s + "dir" + s + "file"));
+  TEST_EQUAL(".", FileManager::GetDirectoryFromFilePath("dir" + ns + "file"));
+  TEST_EQUAL("C:" + s + "root" + s + "dir" + s,
```

Comment:
Yes. Will leave it here for now before I have a real chance to run and improve everything under Windows.

> On Jun 10, 2015, at 20:13, Dima <notifications@github.com> wrote:
> 
> In tests/test_messages_queue.cc <https://github.com/KnowSheet/Alohalytics/pull/36#discussion_r32141429>:
> 
> > +    return tmp_file;
> > +  }
> > +#endif
> > +  return std::string();
> > +}
> > +
> > +void Test_GetDirectoryFromFilePath() {
> > +  const std::string s = std::string(1, FileManager::kDirectorySeparator);
> > +  const std::string ns = (s == "/") ? "\\" : "/";
> > +  TEST_EQUAL("", FileManager::GetDirectoryFromFilePath(""));
> > +  TEST_EQUAL(".", FileManager::GetDirectoryFromFilePath("some_file_name.ext"));
> > +  TEST_EQUAL(".", FileManager::GetDirectoryFromFilePath("evil" + ns + "file"));
> > +  TEST_EQUAL("dir" + s, FileManager::GetDirectoryFromFilePath("dir" + s + "file"));
> > +  TEST_EQUAL(s + "root" + s + "dir" + s, FileManager::GetDirectoryFromFilePath(s + "root" + s + "dir" + s + "file"));
> > +  TEST_EQUAL(".", FileManager::GetDirectoryFromFilePath("dir" + ns + "file"));
> > +  TEST_EQUAL("C:" + s + "root" + s + "dir" + s,
> This is hardcore Windows-specific, right?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/KnowSheet/Alohalytics/pull/36/files#r32141429>.
>